client/src/SessionView/PlayerFlag.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import { useParams } from 'react-router-dom'
+import FactionFlag from '../shared/FactionFlag'
+import { useSession } from './queries'
+
+function PlayerFlag(
+  {
+    disabled,
+    factionKey,
+    selected,
+    onClick,
+    width,
+    height,
+    className,
+    borderWidth,
+  },
+  ref,
+) {
+  const { sessionId } = useParams()
+  const { session } = useSession({
```

Comment:
everything inside SessionView is inside SessionContext, so instead of relying on params you can do `const { players } = useSessionContext()`